[HIGH] Fix missing React hooks dependencies

## Description

Several React components are using hooks without declaring them as dependencies:

1. **GitHubSyncDialog.tsx** - Uses  but hook is not in dependency array
2. **LinkedInImportDialog.tsx** - Multiple hooks missing dependencies (lines 7, 9, 68, 157, 160)
3. **VersionHistory.tsx** - Uses hooks without proper dependency tracking

## QAQC Finding

ESLint warning: Component uses hook(s) not in the dependency array. This can cause bugs with stale closures.

## Acceptance Criteria

- [ ] Fix GitHubSyncDialog.tsx - add missing hook dependencies
- [ ] Fix LinkedInImportDialog.tsx - add all missing hook dependencies  
- [ ] Fix VersionHistory.tsx - add missing hook dependencies
- [ ] Verify with 
> resumeai-saas@0.0.0 lint
> eslint . --ext .ts,.tsx


/home/alex/Projects/ResumeAI/components/AccessibleDialog.test.tsx
  234:13  warning  'container' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/ErrorDisplay.tsx
  64:9  warning  'errorInfo' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/ErrorTestPanel.tsx
  4:3  warning  'ErrorType' is defined but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/ExperienceItem.tsx
  25:62  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/components/GitHubSettings.test.tsx
   8:23  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   9:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  17:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  23:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  29:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  39:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  49:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  67:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  80:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  90:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/components/GitHubSyncDialog.tsx
  160:6  warning  React Hook useEffect has a missing dependency: 'checkConnectionStatus'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
  184:6  warning  React Hook useCallback has a missing dependency: 'loadRepositories'. Either include it or remove the dependency array     react-hooks/exhaustive-deps

/home/alex/Projects/ResumeAI/components/LinkedInImportDialog.tsx
    7:3   warning  'importLinkedInProfile' is defined but never used                                                                             @typescript-eslint/no-unused-vars
    9:3   warning  'disconnectLinkedIn' is defined but never used                                                                                @typescript-eslint/no-unused-vars
   68:10  warning  'oauthWindow' is assigned a value but never used                                                                              @typescript-eslint/no-unused-vars
  157:13  warning  'experience' is assigned a value but never used                                                                               @typescript-eslint/no-unused-vars
  160:13  warning  'education' is assigned a value but never used                                                                                @typescript-eslint/no-unused-vars
  253:3   warning  React Hook "useEffect" is called conditionally. React Hooks must be called in the exact same order in every component render  react-hooks/rules-of-hooks

/home/alex/Projects/ResumeAI/components/LinkedInSettings.tsx
  28:14  warning  'error' is defined but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/MemberList.tsx
  1:27  warning  'useCallback' is defined but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/OfferComparison.tsx
  2:50  warning  'JobOffer' is defined but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/ResumePreview.tsx
   6:38  warning  'useCallback' is defined but never used       @typescript-eslint/no-unused-vars
  40:5   warning  'variant' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/Skeleton.test.tsx
  1:32  warning  'vi' is defined but never used          @typescript-eslint/no-unused-vars
  1:36  warning  'beforeEach' is defined but never used  @typescript-eslint/no-unused-vars
  1:48  warning  'afterEach' is defined but never used   @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/StatusBadge.tsx
  18:7  warning  This assigned value is not used in subsequent statements  no-useless-assignment

/home/alex/Projects/ResumeAI/components/StorageWarning.test.tsx
  2:26  warning  'fireEvent' is defined but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/TeamCard.tsx
  3:10  warning  'showSuccessToast' is defined but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/TeamResumeLibrary.tsx
  101:9  warning  'getInitials' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/TemplateCustomizer.tsx
  23:47  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/components/VariantComparison.tsx
  139:51  warning  'idx' is defined but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/VersionHistory.tsx
  22:6  warning  React Hook useEffect has a missing dependency: 'loadVersions'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/home/alex/Projects/ResumeAI/components/WebhookForm.tsx
  107:20  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/components/editor/EditorActions.tsx
  21:10  warning  'getTimeSince' is defined but never used  @typescript-eslint/no-unused-vars
  40:3   warning  'saveStatus' is defined but never used    @typescript-eslint/no-unused-vars
  41:3   warning  'lastSaved' is defined but never used     @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/editor/EducationItem.tsx
  9:62  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/components/editor/EducationSection.tsx
   2:10  warning  'SimpleResumeData' is defined but never used  @typescript-eslint/no-unused-vars
  10:62  warning  Unexpected any. Specify a different type      @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/components/editor/ExperienceItem.tsx
  9:62  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/components/editor/ExperienceSection.tsx
   2:10  warning  'SimpleResumeData' is defined but never used  @typescript-eslint/no-unused-vars
  10:62  warning  Unexpected any. Specify a different type      @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/components/editor/ProjectItem.tsx
  9:60  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/components/editor/ProjectsSection.tsx
   2:10  warning  'SimpleResumeData' is defined but never used  @typescript-eslint/no-unused-vars
  10:60  warning  Unexpected any. Specify a different type      @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/components/editor/SkillsSection.tsx
  2:10  warning  'SimpleResumeData' is defined but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/editor/VersionHistoryDialog.tsx
  7:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/components/skeletons/SettingsSkeleton.test.tsx
  12:13  warning  'container' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/components/skeletons/TeamsSkeleton.test.tsx
  12:13  warning  'container' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/contexts/FeatureFlagsContext.tsx
  1:44  warning  'useEffect' is defined but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/hooks/useAuth.test.ts
    2:27  warning  'waitFor' is defined but never used       @typescript-eslint/no-unused-vars
   49:18  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   49:53  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   64:20  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   68:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   86:20  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  108:20  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  133:20  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/hooks/useAuth.ts
   9:7   warning  'API_URL' is assigned a value but never used                                                                           @typescript-eslint/no-unused-vars
  16:10  warning  'getAuthHeaders' is defined but never used                                                                             @typescript-eslint/no-unused-vars
  82:5   warning  React Hook useCallback has a missing dependency: 'fetchCurrentUser'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/home/alex/Projects/ResumeAI/hooks/useFocus.test.ts
  29:49  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  40:49  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/hooks/useFocus.ts
  22:30  warning  'returnFocus' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/hooks/useFocusTrap.test.ts
  36:51  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  49:51  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  68:51  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  84:51  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  97:51  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/hooks/useGeneratePackage.test.ts
    4:28  warning  'ResumeData' is defined but never used         @typescript-eslint/no-unused-vars
   90:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  112:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  119:15  warning  'response' is assigned a value but never used  @typescript-eslint/no-unused-vars
  142:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  169:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  186:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  199:29  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  206:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  217:56  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  234:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  255:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  259:96  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  270:29  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  279:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  297:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  316:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  329:29  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  345:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  365:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  390:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  400:28  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  426:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  451:29  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  569:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  582:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  595:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  603:29  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  615:17  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/hooks/useGeneratePackage.ts
    2:33  warning  'useEffect' is defined but never used     @typescript-eslint/no-unused-vars
  218:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  252:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  277:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  316:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/hooks/useTheme.test.ts
   12:30  warning  Unexpected any. Specify a different type         @typescript-eslint/no-explicit-any
   16:20  warning  Unexpected any. Specify a different type         @typescript-eslint/no-explicit-any
   17:7   warning  'getItemSpy' is assigned a value but never used  @typescript-eslint/no-unused-vars
   17:19  warning  Unexpected any. Specify a different type         @typescript-eslint/no-explicit-any
   18:7   warning  'setItemSpy' is assigned a value but never used  @typescript-eslint/no-unused-vars
   18:19  warning  Unexpected any. Specify a different type         @typescript-eslint/no-explicit-any
   19:21  warning  Unexpected any. Specify a different type         @typescript-eslint/no-explicit-any
  116:10  warning  Unexpected any. Specify a different type         @typescript-eslint/no-explicit-any
  129:10  warning  Unexpected any. Specify a different type         @typescript-eslint/no-explicit-any
  144:70  warning  Unexpected any. Specify a different type         @typescript-eslint/no-explicit-any
  161:70  warning  Unexpected any. Specify a different type         @typescript-eslint/no-explicit-any
  164:13  warning  'unmount' is assigned a value but never used     @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/hooks/useVariants.test.ts
   29:22  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   39:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   51:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   67:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   81:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  102:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  110:42  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  118:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  133:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  144:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  155:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  173:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  186:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/hooks/useVariants.ts
  41:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/pages/Billing.tsx
  11:3  warning  'upgradeSubscription' is defined but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/pages/Dashboard.tsx
  77:38  warning  'idx' is defined but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/pages/Editor.bench.test.tsx
   1:24  warning  'vi' is defined but never used                                @typescript-eslint/no-unused-vars
   3:17  warning  'useState' is defined but never used                          @typescript-eslint/no-unused-vars
   9:7   warning  'generateLargeResumeData' is assigned a value but never used  @typescript-eslint/no-unused-vars
  35:24  warning  'count' is defined but never used                             @typescript-eslint/no-unused-vars
  42:13  warning  'container' is assigned a value but never used                @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/pages/Editor.test.tsx
    6:26  warning  'createMockResume' is defined but never used          @typescript-eslint/no-unused-vars
    6:44  warning  'emptyResumeData' is defined but never used           @typescript-eslint/no-unused-vars
   14:9   warning  'mockOnUpdate' is assigned a value but never used     @typescript-eslint/no-unused-vars
  168:13  warning  'emptyData' is assigned a value but never used        @typescript-eslint/no-unused-vars
  179:13  warning  'emptySkillsData' is assigned a value but never used  @typescript-eslint/no-unused-vars
  190:13  warning  'minimalData' is assigned a value but never used      @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/pages/Editor.tsx
  223:21  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/pages/InterviewPractice.tsx
   3:10  warning  'InterviewQuestion' is defined but never used        @typescript-eslint/no-unused-vars
  27:10  warning  'isRecording' is assigned a value but never used     @typescript-eslint/no-unused-vars
  27:23  warning  'setIsRecording' is assigned a value but never used  @typescript-eslint/no-unused-vars
  28:44  warning  Unexpected any. Specify a different type             @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/pages/JobApplications.test.tsx
  96:39  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/pages/JobApplications.tsx
    2:10  warning  'JobApplication' is defined but never used                                                                             @typescript-eslint/no-unused-vars
   13:3   warning  'createJobApplication' is defined but never used                                                                       @typescript-eslint/no-unused-vars
   14:3   warning  'updateJobApplication' is defined but never used                                                                       @typescript-eslint/no-unused-vars
   15:3   warning  'deleteJobApplication' is defined but never used                                                                       @typescript-eslint/no-unused-vars
   16:3   warning  'getApplicationStats' is defined but never used                                                                        @typescript-eslint/no-unused-vars
   37:7   warning  'mapDisplayToApiStatus' is assigned a value but never used                                                             @typescript-eslint/no-unused-vars
  115:10  warning  'showAddModal' is assigned a value but never used                                                                      @typescript-eslint/no-unused-vars
  233:6   warning  React Hook useCallback has a missing dependency: 'sampleResumeData'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
  267:6   warning  React Hook useCallback has a missing dependency: 'sampleResumeData'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/home/alex/Projects/ResumeAI/pages/ResumeManagement.tsx
    1:38  warning  'useCallback' is defined but never used                                                                         @typescript-eslint/no-unused-vars
   21:21  warning  'setFilterTag' is assigned a value but never used                                                               @typescript-eslint/no-unused-vars
   30:10  warning  'pendingOperation' is assigned a value but never used                                                           @typescript-eslint/no-unused-vars
   35:6   warning  React Hook useEffect has a missing dependency: 'loadResumes'. Either include it or remove the dependency array  react-hooks/exhaustive-deps
  368:30  warning  'selected' is defined but never used                                                                            @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/pages/SalaryResearch.tsx
    2:10  warning  'Route' is defined but never used             @typescript-eslint/no-unused-vars
   11:3   warning  'updatePriorities' is defined but never used  @typescript-eslint/no-unused-vars
  339:60  warning  Unexpected any. Specify a different type      @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/pages/Settings.test.tsx
    2:26  warning  'fireEvent' is defined but never used               @typescript-eslint/no-unused-vars
    2:37  warning  'waitFor' is defined but never used                 @typescript-eslint/no-unused-vars
   36:10  warning  Unexpected any. Specify a different type            @typescript-eslint/no-explicit-any
  598:13  warning  'lastNameLabel' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/pages/Settings.tsx
   58:19  warning  'idx' is defined but never used                         @typescript-eslint/no-unused-vars
  212:10  warning  'webhookRefreshKey' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/pages/Teams.tsx
  35:10  warning  'isLoadingMembers' is assigned a value but never used                                                               @typescript-eslint/no-unused-vars
  51:6   warning  React Hook useEffect has a missing dependency: 'loadTeams'. Either include it or remove the dependency array        react-hooks/exhaustive-deps
  58:6   warning  React Hook useEffect has a missing dependency: 'loadTeamDetails'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

/home/alex/Projects/ResumeAI/pages/Webhooks.tsx
  14:10  warning  'webhookRefreshKey' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/pages/Workspace.tsx
    4:10  warning  'SimpleResumeData' is defined but never used   @typescript-eslint/no-unused-vars
    4:28  warning  'ResumeMetadata' is defined but never used     @typescript-eslint/no-unused-vars
    9:8   warning  'WorkspaceSkeleton' is defined but never used  @typescript-eslint/no-unused-vars
   10:10  warning  'getResume' is defined but never used          @typescript-eslint/no-unused-vars
   86:31  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
   87:31  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  204:60  warning  Unexpected any. Specify a different type       @typescript-eslint/no-explicit-any
  236:28  warning  'node' is defined but never used               @typescript-eslint/no-unused-vars
  242:28  warning  'node' is defined but never used               @typescript-eslint/no-unused-vars
  248:28  warning  'node' is defined but never used               @typescript-eslint/no-unused-vars
  251:27  warning  'node' is defined but never used               @typescript-eslint/no-unused-vars
  254:28  warning  'node' is defined but never used               @typescript-eslint/no-unused-vars
  257:28  warning  'node' is defined but never used               @typescript-eslint/no-unused-vars
  258:28  warning  'node' is defined but never used               @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/src/hooks/useFeatureFlags.ts
  39:17  warning  'setError' is assigned a value but never used  @typescript-eslint/no-unused-vars
  61:14  warning  'err' is defined but never used                @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/src/hooks/useStorageQuota.ts
  2:27  warning  'checkStorageWarning' is defined but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/src/lib/accessibility.ts
   6:14  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  62:46  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  66:32  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/src/lib/storage.test.ts
   1:55  warning  'vi' is defined but never used                 @typescript-eslint/no-unused-vars
  11:7   warning  'testData' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/src/lib/storage.ts
  75:24  warning  Do not access Object.prototype method 'hasOwnProperty' from target object  no-prototype-builtins

/home/alex/Projects/ResumeAI/store/store.ts
   7:3   warning  'saveResumeData' is defined but never used  @typescript-eslint/no-unused-vars
  11:10  warning  'TokenManager' is defined but never used    @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/utils/api-client-retry.test.ts
  11:17  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/api-client.test.ts
   69:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   85:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   98:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  107:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  120:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  121:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  129:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  130:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  139:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  147:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  148:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  149:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/api-client.ts
   123:7   warning  'lastResponse' is assigned a value but never used  @typescript-eslint/no-unused-vars
  1007:50  warning  Unexpected any. Specify a different type           @typescript-eslint/no-explicit-any
  1016:42  warning  Unexpected any. Specify a different type           @typescript-eslint/no-explicit-any
  1043:9   warning  'data' is assigned a value but never used          @typescript-eslint/no-unused-vars
  1052:54  warning  Unexpected any. Specify a different type           @typescript-eslint/no-explicit-any
  1056:35  warning  'id' is defined but never used                     @typescript-eslint/no-unused-vars
  1097:28  warning  Unexpected any. Specify a different type           @typescript-eslint/no-explicit-any
  1112:28  warning  Unexpected any. Specify a different type           @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/enhancements/ai-suggestions.ts
  167:47  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  183:13  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  197:49  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  231:65  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  255:54  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  270:48  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/enhancements/version-comparison.ts
   27:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   28:16  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   81:13  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   82:13  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  164:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  165:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/errorHandler.test.ts
    8:8   warning  'ErrorContext' is defined but never used  @typescript-eslint/no-unused-vars
  269:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/errorHandler.ts
    8:3   warning  'getErrorSuggestion' is defined but never used            @typescript-eslint/no-unused-vars
    9:3   warning  'isErrorRetryable' is defined but never used              @typescript-eslint/no-unused-vars
   22:28  warning  Unexpected any. Specify a different type                  @typescript-eslint/no-explicit-any
   69:22  warning  Unexpected any. Specify a different type                  @typescript-eslint/no-explicit-any
   69:52  warning  Unexpected any. Specify a different type                  @typescript-eslint/no-explicit-any
   90:29  warning  Unexpected any. Specify a different type                  @typescript-eslint/no-explicit-any
   90:69  warning  Unexpected any. Specify a different type                  @typescript-eslint/no-explicit-any
  127:11  warning  This assigned value is not used in subsequent statements  no-useless-assignment
  262:13  warning  'payload' is assigned a value but never used              @typescript-eslint/no-unused-vars
  289:28  warning  Unexpected any. Specify a different type                  @typescript-eslint/no-explicit-any
  304:13  warning  Unexpected any. Specify a different type                  @typescript-eslint/no-explicit-any
  305:13  warning  Unexpected any. Specify a different type                  @typescript-eslint/no-explicit-any
  314:13  warning  Unexpected any. Specify a different type                  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/export.test.ts
   58:13  warning  'mockAppendChild' is assigned a value but never used  @typescript-eslint/no-unused-vars
   59:13  warning  'mockRemoveChild' is assigned a value but never used  @typescript-eslint/no-unused-vars
   71:47  warning  Unexpected any. Specify a different type              @typescript-eslint/no-explicit-any
  139:47  warning  Unexpected any. Specify a different type              @typescript-eslint/no-explicit-any
  178:47  warning  Unexpected any. Specify a different type              @typescript-eslint/no-explicit-any
  220:47  warning  Unexpected any. Specify a different type              @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/export.ts
  31:3  warning  'options' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/utils/fetch-timeout.test.ts
   34:40  warning  Unexpected any. Specify a different type        @typescript-eslint/no-explicit-any
   44:13  warning  'timeoutId' is assigned a value but never used  @typescript-eslint/no-unused-vars
   44:40  warning  Unexpected any. Specify a different type        @typescript-eslint/no-explicit-any
   48:29  warning  Unexpected any. Specify a different type        @typescript-eslint/no-explicit-any
  110:28  warning  Unexpected any. Specify a different type        @typescript-eslint/no-explicit-any
  111:16  warning  'e' is defined but never used                   @typescript-eslint/no-unused-vars
  120:19  warning  Unexpected any. Specify a different type        @typescript-eslint/no-explicit-any
  123:46  warning  Unexpected any. Specify a different type        @typescript-eslint/no-explicit-any
  175:13  warning  'spy' is assigned a value but never used        @typescript-eslint/no-unused-vars
  180:29  warning  Unexpected any. Specify a different type        @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/fetch-timeout.ts
  24:18  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  35:36  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  38:27  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  96:39  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/import.test.ts
   12:10  warning  'ResumeData' is defined but never used    @typescript-eslint/no-unused-vars
   15:23  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  414:58  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  418:12  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/linkedin.test.ts
    9:10  warning  'ResumeData' is defined but never used    @typescript-eslint/no-unused-vars
   11:23  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  233:12  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  248:12  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  304:12  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  312:26  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  349:12  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/linkedin.ts
   13:54  warning  'Location' is defined but never used        @typescript-eslint/no-unused-vars
   13:64  warning  'Profile' is defined but never used         @typescript-eslint/no-unused-vars
   51:37  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
   69:50  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
   84:57  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  163:56  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  232:56  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  293:44  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  308:45  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  323:38  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  334:44  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  346:54  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  361:42  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  388:45  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  406:43  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  421:37  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  436:45  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  457:43  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  472:37  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  553:10  warning  'generateId' is defined but never used      @typescript-eslint/no-unused-vars
  678:44  warning  Unexpected any. Specify a different type    @typescript-eslint/no-explicit-any
  748:11  warning  'WorkExperience' is defined but never used  @typescript-eslint/no-unused-vars
  762:11  warning  'Education' is defined but never used       @typescript-eslint/no-unused-vars
  774:11  warning  'Language' is defined but never used        @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/utils/logger.test.ts
  6:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  7:23  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  8:23  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  9:24  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/logger.ts
   26:20  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   27:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   28:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   29:20  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   74:20  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
   75:31  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  147:51  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  171:28  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  180:27  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  189:27  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  198:28  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/prefetch.ts
   5:49  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  12:72  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/retryLogic.test.ts
   12:3   warning  'RetryConfig' is defined but never used   @typescript-eslint/no-unused-vars
  105:9   warning  'mockFetch' is defined but never used     @typescript-eslint/no-unused-vars
  105:20  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  106:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  202:23  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  210:19  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/retryLogic.ts
  102:7  warning  'lastResponse' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/utils/security.ts
   67:7   warning  'deleteCookie' is assigned a value but never used  @typescript-eslint/no-unused-vars
   74:14  warning  '_token' is defined but never used                 @typescript-eslint/no-unused-vars
   94:20  warning  '_token' is defined but never used                 @typescript-eslint/no-unused-vars
  100:24  warning  '_token' is defined but never used                 @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/utils/storage.test.ts
  42:14  warning  'error' is defined but never used  @typescript-eslint/no-unused-vars
  51:14  warning  'error' is defined but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/utils/storage.ts
   7:26  warning  'checkQuotaAvailable' is defined but never used  @typescript-eslint/no-unused-vars
  56:12  warning  'e' is defined but never used                    @typescript-eslint/no-unused-vars
  91:14  warning  'managerError' is defined but never used         @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/utils/toast.test.ts
    2:17  warning  'ToastOptions' is defined but never used  @typescript-eslint/no-unused-vars
  130:41  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/utils/versioning.test.ts
  10:39  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

/home/alex/Projects/ResumeAI/vite.config.ts
  7:9  warning  'env' is assigned a value but never used  @typescript-eslint/no-unused-vars

/home/alex/Projects/ResumeAI/vitest.setup.ts
   9:12  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any
  59:14  warning  Unexpected any. Specify a different type  @typescript-eslint/no-explicit-any

✖ 348 problems (0 errors, 348 warnings) - no dependency warnings

## Files

- 
- 
- 

## Related

ESLint output shows ~30 unused variable warnings and ~60 excessive  type warnings.

--- a/components/GitHubSyncDialog.tsx
+++ b/components/GitHubSyncDialog.tsx
@@ -157,7 +157,7 @@ export const GitHubSyncDialog: React.FC<GitHubSyncDialogProps> = ({
     if (isOpen) {
       checkConnectionStatus();
     }
-  }, [isOpen]);
+  }, [isOpen, checkConnectionStatus]);
 
   // Check connection status
   const checkConnectionStatus = useCallback(async () => {
@@ -181,7 +181,7 @@ export const GitHubSyncDialog: React.FC<GitHubSyncDialogProps> = ({
     } finally {
       setIsLoadingStatus(false);
     }
-  }, []);
+  }, [loadRepositories]);
 
   // Load repositories from GitHub
   const loadRepositories = useCallback(async () => {

--- a/components/LinkedInImportDialog.tsx
+++ b/components/LinkedInImportDialog.tsx
@@ -76,8 +76,6 @@ export const LinkedInImportDialog: React.FC<LinkedInImportDialogProps> = ({
   const [editedSummary, setEditedSummary] = useState('');
   const [editedSkills, setEditedSkills] = useState<string[]>([]);
 
-  if (!isOpen) return null;
-
   const getStepIndex = () => STEPS.findIndex((s) => s.id === currentStep);
 
   // Handle OAuth flow
@@ -393,6 +391,9 @@ export const LinkedInImportDialog: React.FC<LinkedInImportDialogProps> = ({
     e.stopPropagation();
     folderInputRef.current?.click();
   };
+
+  // Conditional render - must be after all hooks
+  if (!isOpen) return null;
 
   return (
     <div className="fixed inset-0 bg-black/50 flex items-center justify-center z-50 animate-in fade-in duration-200">

--- a/components/VersionHistory.tsx
+++ b/components/VersionHistory.tsx
@@ -1,4 +1,4 @@
-import React, { useState, useEffect } from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import { ResumeVersion } from '../types';
 import { listResumeVersions, restoreResumeVersion } from '../utils/api-client';
 import { getVersionTimeAgo, formatVersionNumber } from '../utils/versioning';
@@ -19,9 +19,9 @@ const VersionHistory: React.FC<VersionHistoryProps> = ({ resumeId, onRestore }) 
 
   useEffect(() => {
     loadVersions();
-  }, [resumeId]);
+  }, [resumeId, loadVersions]);
 
-  const loadVersions = async () => {
+  const loadVersions = useCallback(async () => {
     try {
       setLoading(true);
       const data = await listResumeVersions(resumeId);
@@ -32,7 +32,7 @@ const VersionHistory: React.FC<VersionHistoryProps> = ({ resumeId, onRestore }) 
     } finally {
       setLoading(false);
     }
-  };
+  }, [resumeId]);
 
   const handleRestore = async (version: ResumeVersion) => {
     if (

